### DISCUSSION TITLE
Where did the "Client SAML Endpoint" field go?

### DISCUSSION THREAD

**Discussion by msakrejda** [keycloak/keycloak#13797] [Archive]
When creating a new client in the old UI (the 18 default), when I select "Add Client" and change "Client Protocol" to "saml", there's a "Client SAML Endpoint" field on the "Add Client" page. In the new UI, this field does not appear. Where can I enter this information? I'm trying to configure the client to act as a SAML identity provider for local development. If I configure the client as I did in the old UI (except this now-missing field), I get a "Invalid redirect uri" error when I open the IdP-initiated SSO URL (before I can even log in). Reading the [release notes](https://www.keycloak.org/docs/latest/release_notes/index.html#keycloak-19-0-0) I don't see any changes in 19 that would break this.

**Comment by msakrejda** [comment]
Not sure if this is relevant, but I noticed the following lines in the Keycloak logs when I load the IdP-initiated SSO URL:

```
2022-08-15 23:43:35,251 ERROR [org.keycloak.protocol.saml.SamlService] (executor-thread-11) SAML assertion consumer url not set up
2022-08-15 23:43:35,261 WARN  [org.keycloak.events] (executor-thread-11) type=LOGIN_ERROR, realmId=f8316b1c-dd1a-4ec3-9676-b4019108e32b, clientId=null, userId=null, ipAddress=172.17.0.1, error=invalid_redirect_uri
```

**Comment by ssilvert** [comment]
The old UI was a little strange.  Upon creation, the field was called, "Client SAML Endpoint".  But when you go to edit that endpoint, the field was called "Master SAML Processing URL".

In the new UI, we don't let you specify that field in the creation wizard.  Perhaps we should.

But "Master SAML Processing URL" is there if you want to use it.

**Reply by msakrejda** [reply]
That did the trick--thanks for the quick answer! I know this is not strictly related to the new UI (which is great, by the way--I think the new organization and styling is a big improvement), but do you think a better error message is possible if that field is not set and someone visits the IdP-initiated SSO URL? The error is pretty opaque right now. Should I file an issue?

**Reply by ssilvert** [reply]
Sure.  Go ahead and open an enhancement request with all the details at https://github.com/keycloak/keycloak-ui/issues/new/choose
